src/controllers/user.js

```diff
@@ -122,3 +123,31 @@ export const deleteUserById = async (req, res) => {
     return sendDataResponse(res, 500, { error: ERR.DELETE_GENERIC_ERROR })
   }
 }
+
+export const searchUser = async (req, res) => {
```

Comment:
this functionality should be done in the getAll controller

Reply:
Should I chenge the query parameter in the getAll to use my param(name)? or just use a conditional statement?